src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -89,6 +89,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
                     message.PostLogoutRedirectUri = Options.PostLogoutRedirectUri;
                 }
 
+                var ticket = await HandleAuthenticateOnceAsync();
+                if (ticket != null)
+                {
+                    message.IdTokenHint = ticket.Principal?.FindFirst(OpenIdConnectParameterNames.IdToken)?.Value;
```

Comment:
Actually, `RedirectToIdentityProvider` is already gone... replaced by `RedirectToAuthenticationEndpoint` and `RedirectToEndSessionEndpoint` :smile: